Library/Homebrew/macos_version.rb

```diff
@@ -66,6 +66,19 @@ def <=>(other)
     result
   end
 
+  sig { params(comparator: String, other: T.any(Version, T::Array[MacOSVersion])).returns(T::Boolean) }
+  def compare(comparator, other)
+    if other.is_a?(Version)
+      super
+    else
+      case comparator
+      when "==" then other.include?(self)
```

Comment:
==/!= are confusing substitutes for ∈/∉, IMHO (I find `"10.14" == ["10.13", "10.14"]` to be very nonintuitive).

Reply:
It's not a great abstraction, to be sure, but all it's doing is checking if a given system version is in a `depends_on` like this from the `calhash` cask:
```
  depends_on macos: [
    :catalina,
    :big_sur,
    :monterey,
    :ventura,
    :sonoma,
  ]
```
which has this internal representation:
```
#<MacOSRequirement: version=="[#<MacOSVersion: \"10.15\">, #<MacOSVersion: \"11\">, #<MacOSVersion: \"12\">, #<MacOSVersion: \"13\">, #<MacOSVersion: \"14\">]" []>
```

Reply:
Seems nicer to define a different method that calls `#compare` when `other` is not an `Array` instead.

What you seem to want is a check whether an OS dependency is satisfied rather than whether two OS versions are the same.

Reply:
I'll probably need help with that.

Reply:
It may make sense here, mine is more of a driveby comment since I was cc'd (oh, the perils of open source). I haven't worked on `brew` in a minute, though i'm hoping to remedy that, mid-summer or so.

Reply:
@carlocab @dduugg Could you suggest a e.g. `TODO` to add here or provide more help to @EricFromCanada here?

@EricFromCanada do you have any more specific questions we can help with here?

Reply:
If there's a way to refactor this so that `compre` is passed use `include?`/`exclude?` (or `in?`/`not_in`, given the order of arguments), that would be preferred.

If that's messy, I can live with what's here.

Reply:
Changed to adding `allows?` to `MacOSRequirement` instead.